app/addons/fauxton/components.react.jsx

```diff
@@ -84,9 +84,131 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  var Tray = React.createClass({
+
+    getInitialState : function () {
+      return {
+        show: false
+      };
+    },
+
+    toggle : function (done) {
+      if (this.state.show) {
+        this.hide(done);
+      } else {
+        this.show(done);
+      }
+    },
+
+    componentDidMount : function () {
+      var that = this;
+      $('body').on('click.Tray-' + this.props.trayid + "-" + this._rootNodeID, function (e) {
```

Comment:
For Databases I'm with you, but when you want to make the tray reusable, I think it's not the worst idea to make it very sure we're distinct...